SummerSchool/app/src/main/java/nl/rug/www/rugsummerschools/controller/MainActivity.java

```diff
@@ -0,0 +1,386 @@
+    };
+
+    private FragmentStatePagerAdapter mFragmentStatePagerAdapter = new FragmentStatePagerAdapter(getSupportFragmentManager()) {
+        @Override
+        public Fragment getItem(int position) {
+            switch (position) {
+                case 0 : return mFragments[0];
+                case 1 : return mFragments[1];
+                case 2 : return mFragments[2];
+                case 3 : return mFragments[3];
+                case 4 :
+                    if (mAuth.getCurrentUser() == null) {
+                        return mFragments[4];
+                    } else {
+                        return mFragments[5];
+                    }
+                default: return null;
+            }
+        }
+
+        @Override
+        public int getCount() {
+            return FRAGMENTS_SIZE;
+        }
+
+        @Override
+        public int getItemPosition(@NonNull Object object) {
+            if (object instanceof ForumLoginFragment || object instanceof ForumThreadListFragment) {
+                return POSITION_NONE;
+            }
+            return super.getItemPosition(object);
+        }
+    };
+
+    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
+            = new BottomNavigationView.OnNavigationItemSelectedListener() {
+
+        @Override
+        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
+            switch (item.getItemId()) {
+                case R.id.navigation_announcement:
+                    mViewPager.setCurrentItem(0);
+                    return true;
+                case R.id.navigation_general:
+                    mViewPager.setCurrentItem(1);
+                    return true;
+                case R.id.navigation_lecturer:
+                    mViewPager.setCurrentItem(2);
+                    return true;
+                case R.id.navigation_time_table:
+                    mViewPager.setCurrentItem(3);
+                    return true;
+                case R.id.navigation_forum:
+                    mViewPager.setCurrentItem(4);
+                    return true;
+            }
+            return false;
+        }
+    };
+
+    private ViewPager.SimpleOnPageChangeListener mSimpleOnPageChangeListener = new ViewPager.SimpleOnPageChangeListener() {
+        @Override
+        public void onPageSelected(int position) {
+            switch (position) {
+                case 0 :
+                    mNavigation.setSelectedItemId(R.id.navigation_announcement);
+                    break;
+                case 1 :
+                    mNavigation.setSelectedItemId(R.id.navigation_general);
+                    break;
+                case 2 :
+                    mNavigation.setSelectedItemId(R.id.navigation_lecturer);
+                    break;
+                case 3 :
+                    mNavigation.setSelectedItemId(R.id.navigation_time_table);
+                    break;
+                case 4 :
```

Comment:
these would be the same constants as earlier, right?

Reply:
Right.